kpi/tests/api/v2/test_api_submissions.py

```diff
@@ -301,30 +327,43 @@ def test_list_submissions_with_partial_permissions(self):
         partial_perms = {
             PERM_VIEW_SUBMISSIONS: [{'_submitted_by': self.someuser.username}]
         }
-        response = self.client.get(self.submission_url, {"format": "json"})
+        response = self.client.get(self.submission_list_url, {"format": "json"})
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
 
         self.asset.assign_perm(self.anotheruser, PERM_PARTIAL_SUBMISSIONS,
                                partial_perms=partial_perms)
-        response = self.client.get(self.submission_url, {"format": "json"})
+        response = self.client.get(self.submission_list_url, {"format": "json"})
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertTrue(self.asset.deployment.submission_count == 2)
+
+        # Calculate number of submissions that matches partial permissions
+        # condition.
+        all_submissions = self._deployment.get_submissions(self.asset.owner)
+        viewable_submissions_count = 0
+        for submission in all_submissions:
+            if {'_submitted_by': submission['_submitted_by']} in partial_perms[
+                PERM_VIEW_SUBMISSIONS
+            ]:
+                viewable_submissions_count += 1
```

Comment:
Even though you're not completely relying on `deployment.get_submissions()` to do all the filtering, I'd still advocate for not using it at all when you're testing view code that also relies on that same function: correct me if I'm wrong, but the view that services `self.client.get(self.submission_list_url, …)` also eventually calls `deployment.get_submissions()`.

I also far prefer looking for expected IDs than relying on counts.